__tests__/app.test.js

```diff
@@ -118,4 +118,92 @@ describe("GET /api/articles", () => {
         });
       });
   });
+
+  test("200: Filters articles by author", () => {
+    return request(app)
+      .get("/api/articles?author=butter_bridge")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles.length).toBeGreaterThan(0);
+
+        articles.forEach((article) => {
+          expect(article.author).toBe("butter_bridge");
+        });
+      });
+  });
+  test("200: Filters articles by topic", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeInstanceOf(Array);
+        expect(articles.length).toBeGreaterThan(0);
+
+        articles.forEach((article) => {
+          expect(article.topic).toBe("cats");
+        });
+      });
+  });
+  test("400: Responds with an error for invalid sort_by column", () => {
+    return request(app)
+      .get("/api/articles?sort_by=invalid_column")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("bad request");
+      });
+  });
+  test("400: Responds with an error for invalid order query", () => {
+    return request(app)
+      .get("/api/articles?order=invalid_query")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Invalid order query");
+      });
+  });
+});
```

Comment:
Just so you know when you get to that point - the above is covered in a later task! But these tests look good. There is a topic that exists but doesn't have any articles attached to it - you want a 200 and empty array in that case.